src/txgraph.h

```diff
@@ -167,6 +169,34 @@ class TxGraph
      *  so CompareChunks is usable without type-conversion. */
     virtual std::pair<std::vector<FeeFrac>, std::vector<FeeFrac>> GetMainStagingDiagrams() noexcept = 0;
 
+    /** Interface returned by GetBlockBuilder. */
+    class BlockBuilder
+    {
+    protected:
+        /** The next chunk, in topological order plus feerate, or std::nullopt if done. */
+        std::optional<std::pair<std::span<Ref*>, FeePerWeight>> m_current_chunk;
+        /** Make constructor non-public (use TxGraph::GetBlockBuilder()). */
+        BlockBuilder() noexcept = default;
+    public:
+        /** Support safe inheritance. */
+        virtual ~BlockBuilder() = default;
+        /** Determine whether there are more transactions to be included. */
+        explicit operator bool() noexcept { return m_current_chunk.has_value(); }
+        /** Get the chunk that is currently suggested to be included. */
+        const std::span<Ref*>& GetCurrentChunk() noexcept { return m_current_chunk->first; }
+        /** Get the feerate of the currently suggested chunk. */
+        const FeePerWeight& GetCurrentChunkFeerate() noexcept { return m_current_chunk->second; }
```

Comment:
I'm a bit surprised to see `m_current_chunk` as a persistent-ish variable that includes `Ref*` -- that makes it a bug to move `Ref`s around while you have a `BlockBuilder` in existence. I would have thought something more like:

```c++
    struct NextChunk { std::vector<Ref*> refs; FeePerWeight feerate; }
    optional<NextChunk> GetNextChunk();
    void Include(); void Skip();
    
    ...

    while (auto x = bb.GetNextChunk(); x) {
        if (x->feerate.size > remaining) { bb.Skip(); continue; }
        bb.Include();
        mine(x->refs);
    }
```

It's still a bug to move Refs around while you have a `NextChunk`, but at least that's kind-of shorter-lived than a BlockBuilder. I think that would mostly just mean moving the Next() calls into NextChunk().